### PR TITLE
create short directory name in OMEdit temp directory

### PR DESCRIPTION
**Pull request by arun3688**
### Related Issues

https://github.com/OpenModelica/OpenModelica/issues/12171

### Purpose
Currently OMEdit is creating a separate directory using full Qualified path of the models example `"Modelica.Fluid.Examples.HeatExchanger.HeatExchangerSimulation"` and this cause cmake issue with maximum path limit of 250 chars. This PR shorten this directory path to use only the `modelName`  we could get rid of very long directoryPath.

### Example

`C:/Users/DEMO/AppData/Local/Temp/OpenModelica/OMEdit/Modelica.Fluid.Examples.HeatExchanger.HeatExchangerSimulation`
to 
`C:/Users/DEMO/AppData/Local/Temp/OpenModelica/OMEdit/HeatExchangerSimulation`

### CODE REVIEW

**Review by adeas31 — APPROVED** [review]

**Comment by mahge** [comment]
I don't know how it works, does OMEdit delete the directory or remove the files in it before generating new files?

Otherwise there is, albeit low, probability of conflicts since the `modelName` is not unique (even in one library). 

We can maybe hash the qualified name and make that a directory inside the `modelName` directory? If there is no better option.

**Comment by adeas31** [comment]
> I don't know how it works, does OMEdit delete the directory or remove the files in it before generating new files?

OMEdit has 2 options,

`Options->Simulation->Delete intermediate compilation files` the default value for this is `true`. This clears the all generated *.c and *.o files except for actual simulation executable and some needed auxiliary files like `model_init.xml`, `model_info.json` etc.

`Options->Simulation->Delete entire simulation directory of the model when OMEdit is closed` the default value for this is `false`. This makes sure that all the directories that OMEdit generate for simulation are deleted on exit.

> We can maybe hash the qualified name and make that a directory inside the modelName directory? If there is no better option.

Yes, we can add the hash suffix for `modelName`.

**Comment by adeas31** [comment]
> `Options->Simulation->Delete intermediate compilation files` the default value for this is `true`. This clears the all generated *.c and *.o files except for actual simulation executable and some needed auxiliary files like `model_init.xml`, `model_info.json` etc.
> 
> `Options->Simulation->Delete entire simulation directory of the model when OMEdit is closed` the default value for this is `false`. This makes sure that all the directories that OMEdit generate for simulation are deleted on exit.

Aaahhhh...... sorry these options are for simulation. I see now that we are doing it for FMU export.

File: OMEdit/OMEditLIB/MainWindow.cpp [hunk 1]
```diff
@@ -1161,8 +1161,11 @@ void MainWindow::exportModelFMU(LibraryTreeItem *pLibraryTreeItem)
   // show the progress bar
   mpProgressBar->setRange(0, 0);
   showProgressBar();
-  // create a folder with model name to dump the files in it.
-  QString modelDirectoryPath = QString("%1/%2").arg(OptionsDialog::instance()->getGeneralSettingsPage()->getWorkingDirectory(), pLibraryTreeItem->getNameStructure());
+  /*
+   * create a folder with hashed string from model name
+   * see https://github.com/OpenModelica/OpenModelica/issues/12171, to have short temp directory path and dump the files in it.
+  */
+  QString modelDirectoryPath = QString("%1/%2").arg(OptionsDialog::instance()->getGeneralSettingsPage()->getWorkingDirectory(), StringHandler::generateHashPrefix(pLibraryTreeItem->getNameStructure()));
```
[the remark this answers is not in the document]
Comment (arun3688) [reply OMEdit/OMEditLIB/MainWindow.cpp]: @adeas31 all the issues are fixed with this https://github.com/OpenModelica/OpenModelica/pull/12428/commits/ebc7702cfb6e60dc94c404b5fa99189f39ceefb1

**Pull request merged**